pkg/fieldpath/fieldpath.go

```diff
@@ -22,6 +22,15 @@ import (
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/meta"
 )
 
+// formatMap formats map[string]string to a string.
+func formatMap(m map[string]string) string {
+	var l string
+	for key, value := range m {
+		l += key + "=" + value + "\n"
```

Comment:
One other thing that's been discussed is a versioned API for certain types of fields (context: versioned schema for network settings via this API)